### ISSUE TITLE
RBE Action Failures Don't show Error Message

### ISSUE THREAD

**Issue by jlaxson** [bazelbuild/bazel#15206]
### Description of the bug:

In Bazel 5.0.0, when an action fails, its stdout/stderr is shown in the terminal.  In Bazel 5.1.0, there is an error message instead: `Invalid action cache entry c0a87968c7fcdb0db0f4eadbf63f2b2162b9591e18dc1891360bef4a9e77c7fa: expected output <file> does not exist.`

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

BUILD:
```python
genrule(name="failing_rule", outs=["file.txt"], cmd="echo show me the error!; exit 1")
```

`bazel build //package:failing_rule`

Under 5.0.0:
```
ERROR: /package/BUILD:21:8: Executing genrule :failing_rule failed: (Exit 1): bash failed: error executing command /bin/bash -c 'source external/bazel_tools/tools/genrule/genrule-setup.sh; echo show me the error!; exit 1'
show me the error!
```

Under 5.1.0:
```
ERROR: /package/BUILD:21:8: Executing genrule :failing_rule failed: (Exit 34): Invalid action cache entry 97e40eca0142883a7bf97974103e483299bcfd11ebd717a8ffecdc89c834bbc1: expected output libs/android/file.txt does not exist.
```

### Which operating system are you running Bazel on?

darwin arm64

### What is the output of `bazel info release`?

release 5.1.0

### Any other information, logs, or outputs that you want to share?

#15085 seems to have modified the code generating the error message in RemoteExecutionService.java:1027.  The comments on that method imply that this should only happen in --remote_download_minimal, however it happens for me in any of --remote_download_outputs={all/toplevel/minimal}.  Even if intended, this makes debugging failed actions very tedious.

cc @brentleyjones

**Comment by brentleyjones** [comment]
cc: @coeuvre 

Is this fixed in 5.1.1?

**Comment by jlaxson** [comment]
You're right... I missed the boat that 5.1.1 was latest.  All good.

**Comment by kjlubick** [comment]
I am still observing the missing error messages in 5.2.0

**Comment by tharakadesilva** [comment]
Still observe the same behavior in `6.0.0-pre.20221020.1`.

**Comment by Ryang20718** [comment]
I am also observing this missing error in 5.3.1

**Comment by coeuvre** [comment]
Printing the missing error is expected if the declared outputs are not created. 

The problem for this issue was the underlying error message from the action's `stdout/stderr` didn't get printed. In this case `show me the error!`. I believe this issue was fixed in `5.1.1` on onwards.
